packages/waku/src/server.ts

```diff
@@ -20,47 +20,49 @@ export type BuildConfig = {
   customData?: unknown; // should be serializable with JSON.stringify
 }[];
 
-export type RenderEntries = (
+export type RenderEntries<Path extends string> = (
   input: string,
   options: {
     params: unknown | undefined;
-    buildConfig: BuildConfig | undefined;
+    buildConfig: BuildConfig<Path> | undefined;
```

Comment:
Ahhh, I'm planning to remove buildConfig from here.
Can you try anything (I mean type assertion) without modifying `server.ts` and `define-router.ts` for now?

Reply:
Since you removed the buildConfig, are these safe to modify now?

Reply:
I still want to see them unchanged. If some capabilities are missing, can we discuss in advance?

Reply:
Ok, so I was able to greatly simplify the code that I have so far and added a temporary `test.tsx` that is simply to demo the current behavior of the type inferences that are set up thus far.

So I pulled the callback that is meant to be passed into `createPages` to show that `makeEntries` properly infers the types of each string literal path:

![image](https://github.com/user-attachments/assets/85e64dfd-bcc8-4e78-a42b-0d30ecd13329)

This is great, but not quite where I'd like to have the inference yet. This is where I'd like to carry the inference through the defined Router so that the final paths can be inferred through `typeof router` at the end. You can see currently the type of the router is not generic:

![image](https://github.com/user-attachments/assets/67639638-8610-4a83-b2ca-0c9ee39c5aa0)

Reply:
I realized something... I don't need to pass the type through `unstable_defineRouter`... I just need to return anything that has the inferred generic. So, I added `ready` to the return of `createPages`

As you can see, now the type of any path is inferred through the response of the router:

![image](https://github.com/user-attachments/assets/46083c5a-91d3-4eeb-b13b-4a84b38cdfd3)
[file for reference ](https://github.com/dai-shi/waku/pull/854/files#diff-c32c159b88bdd4dc24789fc64dd63421910b3917eec39eb7aa710bc8433d49c7)

I'll plan to extend this to infer the rest of the page type to collect things like the `staticPaths` and render mode in order to generate a final string literal that represents all possible paths for the router to route to.

It's worth acknowledgind that this approach does have the downside of adding to the response body of `createPages` just to get the type of the pages defined, but this seems like a good tradeoff while we do not want to touch `unstable_defineRouter`.

Reply:
> I don't need to pass the type through unstable_defineRouter

That's what I meant by "I mean type assertion".

Reply:
Ah! I get it now... I am asserting that type now as `DO_NOT_USE_ready` - I would happily take naming suggestions on this, I just wanted to get the point across that this type is not for actual use in js.

I have the full page types inferred now 🥳 

![image](https://github.com/user-attachments/assets/a948f4cf-89b9-4d00-b193-6b5000cab13c)

Next steps:
- [ ] Create a union type of all possible paths given the pages
- [ ] Pass this union to an override-able interface so waku projects can consume the pages type safety (same as tanstack router's approach)
- [ ] Use that type in our `<Link />` component
- [ ] Update tests and refactor examples to match new api